### PR TITLE
PLAT-480: Update Python Library to APIv2

### PR DESCRIPTION
**Pull request by phildini**
This PR does a first pass at APIv2 compatibility for the python client library.

All resource files, resource spec files, api files, and api spec files are now auto-generated.

Ran tests locally from python client repo; all passed.

```
platform darwin -- Python 3.6.1, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: /Users/philip/code/patreon-python, inifile: pytest.ini
plugins: cov-2.5.1
collected 65 items

patreon/api_spec.py ........................                                                                                                                                                                                                                                                                                                                         [ 36%]
patreon/oauth_spec.py ..                                                                                                                                                                                                                                                                                                                                             [ 40%]
patreon/jsonapi/parser_spec.py .............                                                                                                                                                                                                                                                                                                                         [ 60%]
patreon/jsonapi/url_util_spec.py ....                                                                                                                                                                                                                                                                                                                                [ 66%]
patreon/schemas/address_spec.py ..                                                                                                                                                                                                                                                                                                                                   [ 69%]
patreon/schemas/benefit_spec.py ..                                                                                                                                                                                                                                                                                                                                   [ 72%]
patreon/schemas/campaign_spec.py ..                                                                                                                                                                                                                                                                                                                                  [ 75%]
patreon/schemas/deliverable_spec.py ..                                                                                                                                                                                                                                                                                                                               [ 78%]
patreon/schemas/goal_spec.py ..                                                                                                                                                                                                                                                                                                                                      [ 81%]
patreon/schemas/media_spec.py .                                                                                                                                                                                                                                                                                                                                      [ 83%]
patreon/schemas/member_spec.py ..                                                                                                                                                                                                                                                                                                                                    [ 86%]
patreon/schemas/oauthclient_spec.py ..                                                                                                                                                                                                                                                                                                                               [ 89%]
patreon/schemas/tier_spec.py ..                                                                                                                                                                                                                                                                                                                                      [ 92%]
patreon/schemas/user_spec.py ..                                                                                                                                                                                                                                                                                                                                      [ 95%]
patreon/schemas/webhook_spec.py ..                                                                                                                                                                                                                                                                                                                                   [ 98%]
patreon/version_compatibility/utc_timezone_spec.py .                                                                                                                                                                                                                                                                                                                 [100%]

---------- coverage: platform darwin, python 3.6.1-final-0 -----------
Name                                                 Stmts   Miss  Cover
------------------------------------------------------------------------
patreon/__init__.py                                      2      0   100%
patreon/api.py                                          94      2    98%
patreon/api_spec.py                                    218     27    88%
patreon/jsonapi/__init__.py                              0      0   100%
patreon/jsonapi/parser.py                               67      4    94%
patreon/jsonapi/parser_spec.py                          70      0   100%
patreon/jsonapi/url_util.py                             15      0   100%
patreon/jsonapi/url_util_spec.py                        15      0   100%
patreon/oauth.py                                        14      0   100%
patreon/oauth_spec.py                                   17      0   100%
patreon/schemas/__init__.py                              0      0   100%
patreon/schemas/address.py                              13      0   100%
patreon/schemas/address_spec.py                         14      0   100%
patreon/schemas/benefit.py                              20      0   100%
patreon/schemas/benefit_spec.py                         14      0   100%
patreon/schemas/campaign.py                             31      0   100%
patreon/schemas/campaign_spec.py                        14      0   100%
patreon/schemas/deliverable.py                           9      0   100%
patreon/schemas/deliverable_spec.py                     14      0   100%
patreon/schemas/goal.py                                  9      0   100%
patreon/schemas/goal_spec.py                            14      0   100%
patreon/schemas/media.py                                14      0   100%
patreon/schemas/media_spec.py                           10      1    90%
patreon/schemas/member.py                               17      0   100%
patreon/schemas/member_spec.py                          14      0   100%
patreon/schemas/oauthclient.py                          17      0   100%
patreon/schemas/oauthclient_spec.py                     14      0   100%
patreon/schemas/tier.py                                 21      0   100%
patreon/schemas/tier_spec.py                            14      0   100%
patreon/schemas/user.py                                 19      0   100%
patreon/schemas/user_spec.py                            14      0   100%
patreon/schemas/webhook.py                              10      0   100%
patreon/schemas/webhook_spec.py                         14      0   100%
patreon/utils.py                                         4      0   100%
patreon/version_compatibility/__init__.py                0      0   100%
patreon/version_compatibility/utc_timezone.py           16     10    38%
patreon/version_compatibility/utc_timezone_spec.py       3      0   100%
------------------------------------------------------------------------
TOTAL                                                  865     44    95%
```

### CODE REVIEW

File: patreon/api.py [hunk 1]
```diff
@@ -16,27 +15,28 @@ def __init__(self, access_token):
...
         return self.__get_jsonapi_doc(
-            build_url('current_user', includes=includes, fields=fields)
+            build_url('identity', includes=includes, fields=fields)
         )
 
-    def fetch_campaign_and_patrons(self, includes=None, fields=None):
-        if not includes:
-            includes = campaign.default_relationships \
-                + [campaign.Relationships.pledges]
+    def fetch_campaigns(self, includes=None, fields=None):
```
Comment (21echoes) [inline patreon/api.py]: this is a second definition of `fetch_campaigns` with the same signature, no?

File: patreon/api.py [hunk 2]
```diff
@@ -16,27 +15,28 @@ def __init__(self, access_token):
...
+        return self.__get_jsonapi_doc(
+            build_url(
+                'campaigns/{0}'.format(campaign_id), includes=includes, fields=fields
+            )
+        )
+
+    def fetch_members(self, campaign_id, page_size, cursor=None, includes=None, fields=None):
+        url = url = 'campaigns/{0}/members'.format(campaign_id)
         params = {'page[count]': page_size}
         if cursor:
```
Comment (21echoes) [inline patreon/api.py]: this code assumes `cursor` will be a datetime object -- is that still the case for paginating through members? https://docs.patreon.com/#get-api-oauth2-v2-campaigns-campaign_id-members implies it is not

**Review by 21echoes — DISMISSED** [review]
one blocker, one potential issue. super excited to get v2 clients out the door!

File: patreon/api.py (hunk 2 above)
Comment (phildini) [reply patreon/api.py]: I don't think we changed the cursor logic at all for APIv2, so we should be fine?

**Review by jrsun — APPROVED** [review]
looks great! just had one question.

**Pull request merged**